### PR TITLE
Adding missing test coverage for category.rb

### PR DESCRIPTION
**Pull request by thovt93**
# Description 

Add tests for the category.rb

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1053

# Screenshots

# Test Coverage

✅

### CODE REVIEW

File: spec/models/category_spec.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+RSpec.describe Category, type: :model do
+  it { is_expected.to validate_presence_of(:user_id) }
```
Comment (jm96441n) [inline spec/models/category_spec.rb]: this should also validate presence of name, as well as test the `self.link` method

File: Gemfile.lock [hunk 2]
```diff
@@ -447,6 +447,8 @@ GEM
       rubyzip (~> 1.2, >= 1.2.2)
     sentry-raven (2.7.4)
       faraday (>= 0.7.6, < 1.0)
+    shoulda-matchers (3.1.2)
```
Comment (julianguyen) [inline Gemfile.lock]: Why is this necessary? Can't we use existing matchers?

**Comment by julianguyen** [comment]
Closing this PR, will finish up #1067 based on the comments and merge it.
